src/main/java/htsjdk/variant/variantcontext/VariantContext.java

```diff
@@ -1233,24 +1232,39 @@ public void validateAlternateAlleles() {
             throw new TribbleException.InternalCodecException(String.format("one or more of the ALT allele(s) for the record at position %s:%d are not observed at all in the sample genotypes", getContig(), getStart()));
     }
 
+    private void validateAttributeIsExpectedSize(final String attributeKey, final int expectedSize ) {
+        final List<Object> actualValues = getAttributeAsList(attributeKey);
+        if (!actualValues.isEmpty()) {
+            // always have at least one actual value
+            final int expectedValuesSize = expectedSize > 0 ? expectedSize : expectedSize + 1;
```

Comment:
Could you change this to `expectedSize > 0 ? expectedSize : 1`  I think that's more clear. 

Since this behavior is pretty specific to these attributes I might change the naming again, i.e. expectedSize -> numAlternateAlleles.

Sorry for the back and forth on this.